lib/util/optional.h

```diff
@@ -0,0 +1,21 @@
+#ifndef LIB_UTIL_OPTIONAL_H
+#define LIB_UTIL_OPTIONAL_H
+
+namespace util {
+
+template <typename T>
+struct Optional {
+  bool isNull;
+  T data;
```

Comment:
These members should be private to help make access to possibly-unset values more obvious through the use of accessors.